pydantic/utils.py

```diff
@@ -534,3 +572,22 @@ def __get__(self, instance: Any, owner: Type[Any]) -> None:
         if instance is None:
             return self.value
         raise AttributeError(f'{self.name!r} attribute of {owner.__name__!r} is class-only')
+
+
+Obj = TypeVar('Obj')
+
+
+def smart_deepcopy(obj: Obj) -> Obj:
```

Comment:
Small food for thought that `smart_deepcopy` seems not particularly descriptive about what this function does, but I cannot come up with a better name so just offering food for thought in case someone else has an idea 😅

Reply:
I also thought about «fast_deepcopy», but function is actually a bit slower in case it has to deepcopy value, so «smart» seemed a better choice.

I agree, it’s kinda vague word, so would love hear any better suggestions :)

Reply:
I agree with both, it's not a perfect name, but I can't think of anything better.